mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseToStdlib.cpp

```diff
@@ -48,2298 +62,939 @@ struct LowerMIOpenOpsStep4Pass
+        composedMaps.push_back(composed);
+        Optional<SmallVector<Value, 8>> init =
+            expandAffineMap(b, loc, composed, op.getUpperInits(i));
+        if (!init.hasValue())
+          return failure();
+        lowerInit.push_back(std::move(*init));
       }
+      lowerInits.push_back(lowerInit);
     }
-  }
-  return ret;
-}
-
-//===----------------------------------------------------------------------===//
-// Utility function to compute various information wrt threadwise_copy.
-// - coordinate length : return value.
-// - composed affine transform.
-// - layered affine transform.
-//===----------------------------------------------------------------------===//
-uint32_t obtainGenericTensorTransformationInfo(
-    const Type type, const ArrayAttr coordTransformsAttr,
-    Optional<AffineMap> &composedTransform,
-    Optional<ArrayAttr> boundsAttr = llvm::None) {
-  // Infer info from a set of transformations.
-  //
-  // 1. If there are coordinate transforms defined, use the input rank of those
-  // 2. Otherwise, use the rank of the affine map applied to the memref
-  //  which is the identiy if unspecified
-  // 3. A bounds attribute overrides this calculation
-  assert(type.isa<MemRefType>() || type.isa<VectorType>());
-  unsigned coordLength = 0;
-  AffineMap typeAffineMap;
-  if (type.isa<MemRefType>()) {
-    MemRefType memrefType = type.cast<MemRefType>();
-    coordLength = memrefType.getRank();
-    if (!memrefType.getLayout().isIdentity())
-      typeAffineMap = memrefType.getLayout().getAffineMap();
-  } else if (type.isa<VectorType>()) {
-    VectorType vectorType = type.cast<VectorType>();
-    coordLength = vectorType.getShape().size();
-    // Vector types doesn't have type-associated affine maps.
-    // Keep typeAffineMaps uninitialized.
-  }
-
-  if (typeAffineMap) {
-    coordLength = typeAffineMap.getNumInputs();
-    composedTransform = typeAffineMap;
-  }
-  // Obtain metadata of coordinate transformations.
-  if (coordTransformsAttr && coordTransformsAttr.size() > 0) {
-    coordLength = coordTransformsAttr[0]
-                      .cast<TransformMapAttr>()
-                      .getMap()
-                      .getValue()
-                      .getNumInputs();
-    SmallVector<AffineMap, 8> maps;
-    for (auto attr : coordTransformsAttr.getAsRange<TransformMapAttr>()) {
-      maps.push_back(attr.getMap().getValue());
-    }
-    composedTransform = composeTransforms(maps);
-  }
-  if (boundsAttr) {
-    coordLength = boundsAttr->size();
-  }
-  // Return computed coordinate length.
-  return coordLength;
-}
-
-//===----------------------------------------------------------------------===//
-// Utility function to emit the logic to copy between naive tensors.
-// This function is used within the lowering logic of threadwise_copy.
-//===----------------------------------------------------------------------===//
-void emitNaiveTensorCopyLogic(OpBuilder &b, Location loc, int64_t nSliceRow,
```

Comment:
Nice this function is getting eliminated. One of the relic porting codes from C++.

Reply:
Yep!